pkg/xds/context/context.go

```diff
@@ -102,7 +102,7 @@ func (mc *MeshContext) ResolveResourceIdentifier(resType core_model.ResourceType
 	var oldestTri *kri.Identifier
 	for _, tri := range mc.BaseMeshContext.DestinationIndex.resolveResourceIdentifiersForLabels(labels) {
```

Comment:
I think `resolveResourceIdentifiersForLabels` should accept `resType` as the first arg. I don't think it's ever correct to use this method for multiple types at once